platform/openide.explorer/src/org/openide/explorer/view/NodeTableModel.java

```diff
@@ -162,10 +162,10 @@ public void setProperties(Property[] props) {
 
                     Object o = props[i].getValue(ATTR_ORDER_NUMBER);
 
-                    if (o instanceof Integer) {
-                        sort.put(new Double(((Integer) o).doubleValue()), Integer.valueOf(ia));
+                    if ((o != null) && o instanceof Integer) {
```

Comment:
The `null` check is unnessary with an `instanceof` check (see discussion above)

The structure can be simplified as in `ide/bugtracking.commons/src/org/netbeans/modules/bugtracking/issuetable/NodeTableModel.java`

Reply:
Fixed.. See above for my comments.